flow/ThreadPrimitives.h

```diff
@@ -44,6 +45,10 @@
 #include <drd.h>
 #endif
 
+// TODO: We should make this dependent on the CPU. Maybe cmake
+// can set this variable properly?
+constexpr size_t CACHE_LINE_SIZE = 64;
+
 class ThreadSpinLock {
```

Comment:
```suggestion
class alignas(CACHE_LINE_SIZE) ThreadSpinLock {
```

Otherwise a spinlock might straddle two cache lines, right?